src/Stryker.Core/Stryker.Core/Reporters/Progress/ConsoleOneLineLogger.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+
+namespace Stryker.Core.Reporters.Progress
+{
+    public interface IConsoleOneLineLogger
+    {
+        void StartLog(string text, params object[] args);
+        void ReplaceLog(string text, params object[] args);
+    }
+
+    public class ConsoleOneLineLogger : IConsoleOneLineLogger
+    {
+        private int _cursorTop;
+
+        private readonly ILogger _logger;
+
+        public ConsoleOneLineLogger(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public void StartLog(string text, params object[] args)
+        {
+            _cursorTop = Console.CursorTop;
+            _logger.LogInformation(text, args);
+        }
+
+        public void ReplaceLog(string text, params object[] args)
+        {
+            var currentCursorTop = Console.CursorTop;
+            var currentCursorLeft = Console.CursorLeft;
+
+            Console.SetCursorPosition(0, _cursorTop);
+            ClearLine();
```

Comment:
I think we could clear only the part till currentCursorLeft. This way not the whole line has to be cleared.

Reply:
I think in that case we would have to keep track of the original cursorLeft position, the line that we want to clear not current one. I would keep it as it is, because I don't think it has so much impact on performance changing it would bring in additional complexity / bugs.